src/redux/gas.ts

```diff
@@ -252,88 +245,11 @@ export const gasUpdateToCustomGasFee = (gasParams: GasFeeParams) => async (dispa
   });
 };
 
-export const getPolygonGasPrices = async () => {
-  try {
-    const {
-      data: {
-        data: { legacy: result },
-      },
-    } = (await rainbowMeteorologyGetData(ChainId.polygon)) as {
-      data: RainbowMeteorologyLegacyData;
-    };
-    const polygonGasPriceBumpFactor = 1.05;
-
-    // Override required to make it compatible with other responses
-    const polygonGasStationPrices = {
-      fast: Math.ceil(Number(multiply(result['proposeGasPrice'], polygonGasPriceBumpFactor))),
-      // 1 blocks, 2.5 - 3 secs
-      fastWait: 0.05,
-      normal: Math.ceil(Number(multiply(result['safeGasPrice'], polygonGasPriceBumpFactor))),
-      // 2 blocks, 6 secs
-      normalWait: 0.1,
-      urgent: Math.ceil(Number(multiply(result['fastGasPrice'], polygonGasPriceBumpFactor))),
-      // 1 blocks, 2.5 - 3 secs
-      urgentWait: 0.05,
-    };
-    return polygonGasStationPrices;
-  } catch (e) {
-    logger.error(new RainbowError(`[redux/gas]: failed to fetch polygon gas prices ${e}`));
-    return null;
-  }
-};
-
-export const getBscGasPrices = async () => {
-  try {
-    const {
-      data: {
-        data: { legacy: result },
-      },
-    } = (await rainbowMeteorologyGetData(ChainId.bsc)) as {
-      data: RainbowMeteorologyLegacyData;
-    };
-
-    const bscGasPriceBumpFactor = 1.05;
-
-    // Override required to make it compatible with other responses
-    const bscGasStationPrices = {
-      fast: Math.ceil(Number(multiply(result['proposeGasPrice'], bscGasPriceBumpFactor))),
-      // 1 blocks, 2.5 - 3 secs
-      fastWait: 0.05,
-      normal: Math.ceil(Number(multiply(result['safeGasPrice'], bscGasPriceBumpFactor))),
-      // 2 blocks, 6 secs
-      normalWait: 0.1,
-      urgent: Math.ceil(Number(multiply(result['fastGasPrice'], bscGasPriceBumpFactor))),
-      // 1 blocks, 2.5 - 3 secs
-      urgentWait: 0.05,
-    };
-    return bscGasStationPrices;
-  } catch (e) {
-    logger.error(new RainbowError(`[redux/gas]: failed to fetch BSC gas prices ${e}`));
-    return null;
-  }
-};
-export const getArbitrumGasPrices = async () => {
-  const provider = getProvider({ chainId: ChainId.arbitrum });
+export const getProviderGasPrices = async ({ chainId }: { chainId: ChainId }) => {
+  const provider = getProvider({ chainId });
   const baseGasPrice = await provider.getGasPrice();
-  const normalGasPrice = weiToGwei(baseGasPrice.toString());
-
-  const priceData = {
-    fast: Number(normalGasPrice),
-    fastWait: 0.14,
-    // 2 blocks, 8 secs
-    normal: Number(normalGasPrice),
-    normalWait: 0.14,
-    urgent: Number(normalGasPrice),
-    urgentWait: 0.14,
-  };
-
-  return priceData;
-};
 
-export const getOptimismGasPrices = async () => {
-  const provider = getProvider({ chainId: ChainId.optimism });
-  const baseGasPrice = await provider.getGasPrice();
-  const normalGasPrice = weiToGwei(baseGasPrice.toString());
+  const normalGasPrice = toHex(Math.ceil(Number((baseGasPrice.toString(), 1.05))));
```

Comment:
not sure what I'm looking at here in `Number((baseGasPrice.toString(), 1.05))`

Reply:
added addBuffer